lms/static/scripts/frontend_apps/utils/test/use-warn-on-page-unload-test.js

```diff
@@ -0,0 +1,62 @@
+import { mount } from 'enzyme';
+import { useState } from 'preact/hooks';
+
+import { useWarnOnPageUnload } from '../use-warn-on-page-unload';
+
+describe('useWarnOnPageUnload', () => {
+  let fakeWindow;
+  const FakeComponent = () => {
+    const [isUnsaved, setUnsaved] = useState(true);
+
+    useWarnOnPageUnload(isUnsaved, fakeWindow);
+
+    return (
+      <button type="button" onClick={() => setUnsaved(false)}>
+        Set not unsaved
+      </button>
+    );
+  };
+  const createComponent = () => mount(<FakeComponent />);
+
+  beforeEach(() => {
+    fakeWindow = new EventTarget();
+    sinon.stub(fakeWindow, 'addEventListener');
+    sinon.stub(fakeWindow, 'removeEventListener');
```

Comment:
I was going to suggest _not_ stubbing these functions. Instead in your tests you can dispatch a `beforeunload` event at the `fakeWindow` and then check `Event.defaultPrevented` and `Event.returnValue`.

Reply:
That makes sense. Will do the changes.

Reply:
For some reason `Event.defaultPrevented` is coming as false, even thought I can see the event listener is being invoked.

I think I have an idea why, but I want to move things forward, so I'm going to merge this as is and create a separate issue to improve this test ASAP.

Reply:
> For some reason Event.defaultPrevented is coming as false, even thought I can see the event listener is being invoked.

Did you set the `cancelable` flag when constructing the event?

Reply:
Ugh, no, that was it. I was going on a completely different direction.

I'll push another PR shortly. Thanks for the tip.

Reply:
There https://github.com/hypothesis/lms/pull/5688